### PR TITLE
Self-Service Docker Images 

### PR DESCRIPTION
**Pull request by etiennedi**
### What's being changed:
* Add GH-action self-service reports with instructions of how to use a preview image.

### Review checklist

- [ ] Documentation has been updated, if necessary. Link to changed documentation:
- [ ] Chaos pipeline run or not necessary. Link to pipeline:
- [ ] All new code is covered by tests where it is reasonable.
- [ ] Performance tests have been run or not necessary.

### CODE REVIEW

File: .github/workflows/pull_requests.yaml [hunk 1]
```diff
@@ -9,93 +9,93 @@ on:
...
+  #     - name: Set up Go
+  #       uses: actions/setup-go@v3
+  #       with:
+  #         go-version: 1.19
+  #         cache: true
+  #     - name: Acceptance tests (modules)
+  #       run: ./test/run.sh --acceptance-only
   Push-Docker:
-    needs: [Acceptance-Tests, Modules-Acceptance-Tests, Unit-Tests, Integration-Tests]
+    # needs: [Acceptance-Tests, Modules-Acceptance-Tests, Unit-Tests, Integration-Tests]
```
Comment (dirkkul) [inline .github/workflows/pull_requests.yaml:98]: remove comments? :)

File: ci/push_docker.sh [hunk 2]
```diff
@@ -23,14 +22,27 @@ function release() {
             return 1
         fi
         tag_exact="${DOCKER_REPO}:${weaviate_version}"
+  else
+    pr_title="$(echo -n "$PR_TITLE" | tr '[:upper:]' '[:lower:]' | tr -c -s '[:alnum:]' '-')"
+    tag_preview="${DOCKER_REPO}:preview-${pr_title}-${git_hash}"
   fi
 
-  args=("--build-arg=GITHASH=$git_hash" "--platform=linux/amd64,linux/arm64" "--target=weaviate" "-t=$tag_latest" "--push")
+  # args=("--build-arg=GITHASH=$git_hash" "--platform=linux/amd64,linux/arm64" "--target=weaviate" "-t=$tag_latest" "--push")
```
Comment (dirkkul) [inline ci/push_docker.sh:30]: reenable

File: ci/push_docker.sh [hunk 3]
```diff
@@ -23,14 +22,27 @@ function release() {
...
+  args=("--build-arg=GITHASH=$git_hash" "--platform=linux/amd64" "--target=weaviate" "-t=$tag_latest" "--push")
   if [ -n "$tag_exact" ]; then
+    # exact tag on master
     args+=("-t=$tag_exact")
   fi
+  if [ -n "$tag_preview" ]; then
+    # preview tag on PR builds
```
Comment (dirkkul) [inline ci/push_docker.sh:37]: also on push? Eg what would happen after a merge to master?

**Review by dirkkul — APPROVED** [review]

File: .github/workflows/pull_requests.yaml (hunk 1 above)
Comment (etiennedi) [reply .github/workflows/pull_requests.yaml:98]: Haha, sorry, I put the status to "Draft" again as I started implementing Andrzej's review feedback. will fix this before marking it as ready again.

File: ci/push_docker.sh (hunk 2 above)
Comment (etiennedi) [reply ci/push_docker.sh:30]: Haha, sorry, I put the status to "Draft" again as I started implementing Andrzej's review feedback. will fix this before marking it as ready again.

File: ci/push_docker.sh (hunk 3 above)
Comment (etiennedi) [reply ci/push_docker.sh:37]: Right now, no report yet. Would like to fix that in a separate PR.

**Comment by etiennedi** [comment]
@dirkkul @aliszka - now it should be ready to review again :-)

File: ci/docker_report.md.tpl [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+## Use at your own risk :warning:
+
+Preview builds make no promises about stability or feature completeness.  Use them at your own risk. A preview build is not generated if tests failed,  so they have at least passed the common test suite. They may or may not have  been subjected to the asynchronous stress test and chaos pipelines.
+
+## Usage :newspaper:
+
+### Docker-compose
+
+You can obtain a [docker-compose.yaml here](https://weaviate.io/developers/weaviate/current/installation/docker-compose.html#configurator) and configure it to your liking. Then make sure to set `services.weaviate.image` to `$PREVIEW_TAG'`. For example, like so:
```
Comment (aliszka) [inline ci/docker_report.md.tpl:17]: \`$PREVIEW_TAG'\`
unnecessary `'`

**Review by aliszka — APPROVED** [review]
👍

File: ci/docker_report.md.tpl (hunk 4 above)
Comment (etiennedi) [reply ci/docker_report.md.tpl:17]: ah, sorry, I missed this. I'll fix it on the merged branches

**Pull request merged**